libs/vkd3d/resource.c

```diff
@@ -2552,6 +2552,16 @@ HRESULT d3d12_resource_create_placed(struct d3d12_device *device, const D3D12_RE
         /* Align manually. This works because we padded the required allocation size reported to the app. */
         VK_CALL(vkGetImageMemoryRequirements(device->vk_device, object->res.vk_image, &memory_requirements));
         heap_offset = align(heap_offset, memory_requirements.alignment);
+    } else {
+        VK_CALL(vkGetBufferMemoryRequirements(device->vk_device, object->res.vk_buffer, &memory_requirements));
```

Comment:
Braces, also as mentioned cannot use vkGetBufferMemoryRequirements here, since we always map to the underlying VkBuffer in the heap allocation. A buffer resource is just offset + range into a heap. So I think it's enough to check `heap_offset + desc->Width <= heap->allocation.resource.size`

Reply:
Oh right, misread your suggestion about vkGetBufferMemoryReqs.